packages/server/lib/util/require_async.ts

```diff
@@ -0,0 +1,88 @@
+import _ from 'lodash'
+import * as path from 'path'
+import * as cp from 'child_process'
+import * as inspector from 'inspector'
+import * as util from '../plugins/util'
+import * as errors from '../errors'
+import Debug from 'debug'
+
+const debug = Debug('cypress:server:require_async')
+
+let requireProcess: cp.ChildProcess | null
+
+interface RequireAsyncOptions{
+  projectRoot: string
+  loadErrorCode: string
+  /**
+   * members of the object returned that are functions and will need to be wrapped
+   */
+  functionNames: string[]
+}
+
+interface ChildOptions{
+  stdio: 'inherit'
+  execArgv?: string[]
+}
+
+const killChildProcess = () => {
+  requireProcess && requireProcess.kill()
+  requireProcess = null
+}
+
+export default async function requireAsync (filePath: string, options: RequireAsyncOptions): Promise<any> {
+  return new Promise((resolve, reject) => {
+    if (requireProcess) {
+      debug('kill existing config process')
+      killChildProcess()
```

Comment:
Do we need to return here or is it fine to continue execution?

Reply:
We could return to clarify the behavior, but this is not needed for the Promise to work

Reply:
We should not return